### PR TITLE
Refactor QVTKRenderWindowInteractor [WIP]

### PR DESCRIPTION
**Pull request by kitchoi**
Attempt to use the vtk.qt4.QVTKRenderWindowInteractor when possible.  It also makes clear the patches we needed.  Some of these patches may be sent upstream to VTK.  However, they are likely to be kept around for sometime unless there are plans for discontinuing supports for older VTK versions.

This would supersede #304 (except for the test case added there).

### CODE REVIEW

File: tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py [hunk 1]
```diff
@@ -271,50 +177,17 @@ def _get_win_id(self):
...
-
-    def sizeHint(self):
-        return QSize(400, 400)
-
-    def paintEngine(self):
-        return None
-
     def paintEvent(self, ev):
+        super(QVTKRenderWindowInteractor, self).paintEvent(ev)
         self._RenderWindow.Render()
```
Comment (kitchoi) [inline tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py]: I am not sure if this `self._RenderWindow.Render()` is only required by tvtk Scene.  If so, then it should be moved to tvtk/pyface/ui/qt4/scene.py

File: tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py [hunk 2]
```diff
@@ -271,50 +177,17 @@ def _get_win_id(self):
     def Finalize(self):
```
Comment (prabhuramachandran) [inline tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py]: IIRC the finalize used to cause a crash on OS X, I can't remember exactly how right now but can you test this version please to ensure that when you create a window and close it, it doesn't crash.  Thanks!

Comment (kitchoi) [inline tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py]: I tried
```
from mayavi import mlab
mlab.figure()
mlab.close()
```
with this branch on OS X 10.11.2, PySide/PyQt with Qt 4.8.7 and there is no crash.
Also run the integratedtests and no crash either.  Note that my OS X version is fairly up-to-date...

File: tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py [hunk 3]
```diff
@@ -174,19 +89,17 @@ def __init__(self, parent=None, wflags=Qt.WindowFlags(), **kw):
             rw = None
 
         # create qt-level widget
-        QWidget.__init__(self, parent, wflags|Qt.MSWindowsOwnDC)
+        QWidget.__init__(self, parent, wflags | Qt.MSWindowsOwnDC)
```
Comment (prabhuramachandran) [inline tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py]: Maybe a comment as to why we are not calling the parent's __init__ would be good.

**Comment by kitchoi** [comment]
I just tested the branch with integrationtests, and it no longer works.  :(  I will have to revisit this.

**Comment by prabhuramachandran** [comment]
@kitchoi -- is it OK to close this now?  The interactor code has also changed a bit now and more inline with the version in VTK.  Please reopen if you need to and the issues are resolved.  Thanks!

**Pull request closed without merging**